aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUserPool.java

```diff
@@ -268,6 +272,12 @@ public CognitoUserPool(Context context, String userPoolId, String clientId, Stri
     public CognitoUserPool(Context context, String userPoolId, String clientId, String clientSecret, ClientConfiguration clientConfiguration, Regions region, String pinpointAppId) {
         initialize(context);
         this.context = context;
+        if (userPoolId.isEmpty() || clientId.isEmpty()) {
+            throw new IllegalArgumentException("Both UserPoolId and ClientId are required. ");
+        }
+        if (userPoolId.length() > USER_POOL_ID_MAX_LENGTH || !Pattern.matches("[\\w-]+_[0-9a-zA-Z]+", userPoolId)) {
```

Comment:
Maybe extract `[\\w-]+_[0-9a-zA-Z]+` as a constant since it's being used in a couple of different places